### ISSUE TITLE
`jest.retryTimes()` falsely reports `PASS` for repeated failing tests

### ISSUE THREAD

**Issue by viktorijasujetaite** [wix/Detox#2040]
**Is your feature request related to a problem? Please describe.**
I want to implement retry mechanism for e2e tests using jest.retryTimes(). 
For tests that should fail 100%, incoherent results are returned - e.g. jest.retryTimes(3) returns statuses FAIL, PASS, PASS, PASS.

**Describe the solution you'd like**
Report test results correctly when using jest.retryTimes().
In advance, specify "PASS" amount in the performed retries that would set final test result as "PASS" 

**Additional context**
When code is modified to print out status with try..catch:
```
it('Assert non-existing text - Fails 100%', async () => {
    try {
      await expect(element(by.text('VIKTORIJA`s TEST'))).toBeVisible();
      console.log('SUCCESS')
    } catch(ex) {
      console.log('failure')
      throw ex
    }
);
```

`failure` is logged, so test does actually fail. This points to Detox jest reporter not understanding those results correctly

**Comment by d4vidi** [comment]
@viktorijasujetaite following this example and by using `jest.retryTimes(3)` with jest-circus I get the following output (latest Detox):
```
detox[18812] INFO:  Sanity is assigned to emulator-13202 (Pixel_API_28)
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100%
detox[18812] INFO:  at e2e/01.sanity.test.js:12:15
 failure
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100% [FAIL]
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100%
detox[18812] INFO:  at e2e/01.sanity.test.js:12:15
 failure
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100% [FAIL]
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100%
detox[18812] INFO:  at e2e/01.sanity.test.js:12:15
 failure
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100% [FAIL]
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100%
detox[18812] INFO:  at e2e/01.sanity.test.js:12:15
 failure
detox[18812] INFO:  Sanity: Assert non-existing text - Fails 100% [FAIL]

 FAIL  e2e/01.sanity.test.js (12.026s)
  Sanity
    ✕ Assert non-existing text - Fails 100% (1397ms)

  ● Sanity › Assert non-existing text - Fails 100%
```

**Comment by d4vidi** [comment]
Appears to be a `Jest` bug. Fixed in jest + jest-circus `25.3.0` (perhaps even `25.0.0`).
